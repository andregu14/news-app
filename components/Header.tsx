import { MaterialCommunityIcons } from "@expo/vector-icons";
import { View, ViewProps } from "./Themed";
import Colors from "@/constants/Colors";
import { useColorScheme } from "@/components/useColorScheme";
import { StyleSheet, TouchableOpacity } from "react-native";

type HeaderProps = ViewProps & {
  onMenuPress?: () => void
  onAccountPress?: () => void
}

export default function Header(props: HeaderProps) {
  const colorScheme = useColorScheme();
  const color = Colors[colorScheme ?? "light"].headerIcon;
  const iconSize = 28;

  return (
    <View style={[props.style, { width: "100%" }]}>
      <View style={styles.container} testID="header">
        <TouchableOpacity onPress={props.onAccountPress} testID="account-button">
          <MaterialCommunityIcons
            name="account"
            size={iconSize}
            color={color}
            testID="icon-account"
          />
        </TouchableOpacity>
        <MaterialCommunityIcons
          name="diamond-stone"
          size={iconSize}
          color={color}
          testID="icon-diamond-stone"
        />
        <TouchableOpacity onPress={props.onMenuPress} testID="menu-button">
          <MaterialCommunityIcons
            name="menu"
            size={iconSize}
            color={color}
            testID="icon-menu"
          />
        </TouchableOpacity>
      </View>
      <View
        style={styles.separator}
        lightColor="#eee"
        darkColor="rgba(255,255,255,0.1)"
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    width: "100%",
    gap: 60,
    justifyContent: "space-around",
  },
  separator: {
    marginTop: 10,
    height: 1,
    width: "100%",
  },
});
