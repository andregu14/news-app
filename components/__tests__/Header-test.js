import { render, screen } from '@testing-library/react-native';
import Header from '../Header';

describe('<Header />', () => {
    it('renders correctly', () => {
        render(<Header />);
        const headerComponent = screen.getByTestId("header")
        expect(headerComponent).toBeVisible();
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with icons', () => {
        render(<Header />);
        const icons = [
            screen.getByTestId('icon-account'),
            screen.getByTestId('icon-diamond-stone'),
            screen.getByTestId('icon-menu')
        ];
        icons.forEach(icon => expect(icon).toBeTruthy());
        expect(screen.toJSON()).toMatchSnapshot();
    });
});
