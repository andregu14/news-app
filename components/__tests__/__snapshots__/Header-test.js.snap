// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        undefined,
        {
          "width": "100%",
        },
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "flexDirection": "row",
          "gap": 60,
          "justifyContent": "space-around",
          "width": "100%",
        },
      ]
    }
    testID="header"
  >
    <Text />
    <Text />
    <Text />
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#eee",
        },
        {
          "height": 1,
          "marginTop": 10,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;

exports[`<Header /> renders correctly with icons 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        undefined,
        {
          "width": "100%",
        },
      ],
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "flexDirection": "row",
          "gap": 60,
          "justifyContent": "space-around",
          "width": "100%",
        },
      ]
    }
    testID="header"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#3F3F3F",
            "fontSize": 28,
          },
          undefined,
          {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon-account"
    >
      󰀄
    </Text>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#3F3F3F",
            "fontSize": 28,
          },
          undefined,
          {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon-diamond-stone"
    >
      󰇈
    </Text>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#3F3F3F",
            "fontSize": 28,
          },
          undefined,
          {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon-menu"
    >
      󰍜
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#eee",
        },
        {
          "height": 1,
          "marginTop": 10,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
